examples/QIR/Optimization/README.md

```diff
@@ -299,3 +299,268 @@ attributes #1 = { "EntryPoint" }
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Quantum.Qir.Runtime" Version="0.18.2106148911-alpha" GeneratePathProperty="true" />
+    <PackageReference Include="Microsoft.Quantum.Simulators" Version="0.18.2106148911" GeneratePathProperty="true" />
+  </ItemGroup>
+
+  <Target Name="GetDependencies" AfterTargets="Build">
+    <PropertyGroup>
+      <QirRuntimeHeaders>$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/any/native/include</QirRuntimeHeaders>
+      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('OSX'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/osx-x64/native</QirRuntimeLibs>
+      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/win-x64/native</QirRuntimeLibs>
+      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('Linux'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/linux-x64/native</QirRuntimeLibs>
+      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('OSX'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/osx-x64/native/Microsoft.Quantum.Simulator.Runtime.dll</SimulatorRuntime>
+      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/win-x64/native/Microsoft.Quantum.Simulator.Runtime.dll</SimulatorRuntime>
+      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('Linux'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/linux-x64/native/Microsoft.Quantum.Simulator.Runtime.dll</SimulatorRuntime>
+    </PropertyGroup>
+    <ItemGroup>
+      <_QirRuntimeLibFiles Include="$(QirRuntimeLibs)/**/*.*" Exclude="$(QirRuntimeLibs)/**/*.exe" />
+      <_QirRuntimeHeaderFiles Include="$(QirRuntimeHeaders)/**/*.hpp" />
+    </ItemGroup>
+    <Copy SourceFiles="$(SimulatorRuntime)" DestinationFolder="$(BuildOutputPath)" SkipUnchangedFiles="true" />
+    <Copy SourceFiles="@(_QirRuntimeLibFiles)" DestinationFolder="$(BuildOutputPath)\%(RecursiveDir)" SkipUnchangedFiles="true" />
+    <Copy SourceFiles="@(_QirRuntimeHeaderFiles)" DestinationFolder="$(BuildOutputPath)\%(RecursiveDir)" SkipUnchangedFiles="true" />
+  </Target>
+
+</Project>
+```
+
+Build the project again with `dotnet build` from the project root directory.
+You should see the following important files appear in a folder named `build`, among others:
+
+```
+build
+├── Microsoft.Quantum.Qir.QSharp.Core.dll
+├── Microsoft.Quantum.Qir.QSharp.Foundation.dll
+├── Microsoft.Quantum.Qir.Runtime.dll
+├── Microsoft.Quantum.Simulator.Runtime.dll
+├── QirContext.hpp
+├── QirRuntime.hpp
+└── SimFactory.hpp
+```
+
+(**Linux**) The `Microsoft.Quantum.Qir.*` dynamic libraries will already have the right naming scheme for Clang to use, but the `Microsoft.Quantum.Simulator.Runtime` library needs to be renamed.
+The proper name format is `lib<library-name>.so`.
+
+Execute the following command from the project root directory:
+
+```bash
+mv build/Microsoft.Quantum.Simulator.Runtime.dll build/libMicrosoft.Quantum.Simulator.Runtime.so
+```
+
```

Comment:
With the merge of the cloud simulator feature branch, we now have a bugfix for this situation (see https://github.com/microsoft/qsharp-runtime/commit/9fa45deda6caeb8785b292ce175a79fe1bc9731f#diff-164e9c49ce0f05da64d60ae5cf3063a3e017f67ef74b0ba6bcb0a62f62506595). The simulator library infrastructure will always look for `Microsoft.Quantum.Simulator.Runtime.dll` first and then fallback to looking for `libMicrosoft.Quantum.Simulator.Runtime.so`. This means the rename is no longer necessary as of today's release (https://docs.microsoft.com/en-us/azure/quantum/qdk-relnotes#qdk-version-0182107153439). So this section can just be removed!

Reply:
Oops... I accidentally clicked merge on this before you had a chance to respond to this comment. I'll make a quick PR removing this part and add you for review!